fourteen-testing-api.js

```diff
@@ -0,0 +1,714 @@
+"use strict";
+
+module.exports = function(file, api) {
+  const j = api.jscodeshift;
+  const root = j(file.source);
+
+  const SETUP_TYPE_METHODS = ["setupComponentTest"];
+
+  function isSetupTypeMethod(nodePath) {
+    return SETUP_TYPE_METHODS.some(name => {
+      let matcher = {
+        type: "ExpressionStatement",
+        expression: { callee: { name } }
+      };
+
+      return j.match(nodePath, matcher);
+    });
+  }
+
+  const GLOBAL_NODE_PATH_CACHE = new Map();
+  // returns startAppAssignment nodePath or false;
+  function findStartAppAssignment(nodePath) {
+
+    let startAppAssignment = GLOBAL_NODE_PATH_CACHE.has(nodePath);
+
+    if (startAppAssignment) { return GLOBAL_NODE_PATH_CACHE.get(nodePath) }
+
+    let assignments = j(nodePath).find(j.ExpressionStatement, { expression: { right: { callee: { name: 'startApp' } } } });
+
+    if (assignments.length >= 1) {
+      let foundAssignment = assignments.paths()[0];
+      GLOBAL_NODE_PATH_CACHE.set(nodePath, foundAssignment);
+      return foundAssignment;
+    } else {
+      return false;
+    }
+  }
+
+  function findDestroyAppCall(nodePath) {
+    let destroys = j(nodePath).find(j.ExpressionStatement, { expression: {callee: {  name: 'destroyApp' } } } )
+
+    if (destroys.length >= 1) {
+      let foundDestroy = destroys.paths()[0];
+      j(foundDestroy).remove();
+    }
+  }
+
+  function processExpressionForApplicationTest(testExpression) {
+    // mark the test function as an async function
+    let testExpressionCollection = j(testExpression);
+    // collect all potential statements to be imported
+    let specifiers = new Set();
+
+    // First - remove andThen blocks
+    removeAndThens(testExpressionCollection);
+
+    // Second - Transform to await visit(), click, fillIn, touch, etc and adds `async` to scope
+    [
+      'visit',
+      'find',
+      'waitFor',
+      'fillIn',
+      'click',
+      'blur',
+      'focus',
+      'tap',
+      'triggerEvent',
+      'triggerKeyEvent',
+    ].forEach(type => {
```

Comment:
Re: the previous comment, not sure about this: if this is supposed to add `await` to the old global test helpers, where do `blur`, `focus`, `tap`, `waitFor` and `triggerKeyEvent` then come from? As there are not global test helpers from `ember-testing`, AFAICT: https://github.com/emberjs/ember.js/blob/master/packages/ember-testing/lib/helpers.js#L16-L23